ha/fault_tolerance/cluster_stop_monitor.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+"""
+Handler for handling cluster stop events received from CSM
+"""
+
+import json
+import ast
+
+from cortx.utils.conf_store import Conf
+from cortx.utils.log import Log
+from ha.util.message_bus import MessageBus, CONSUMER_STATUS, MessageBusConsumer
+
+from ha import const
+from ha.k8s_setup.const import _DELIM
+
+class ClusterStopMonitor:
+
+    def __init__(self):
+        """Init method"""
+        self._consumer = self._get_consumer()
+
+    def start(self):
+        """
+        Start to listen messages.
+        """
+        self._consumer.start()
+
+    def _get_consumer(self) -> MessageBusConsumer:
+        """
+           Returns an object of MessageBusConsumer class which will listen on
+           cluster_event message type and callback will be executed
+        """
+
+        self._consumer_id = Conf.get(const.HA_GLOBAL_INDEX, f'CLUSTER_STOP_MON{_DELIM}consumer_id')
+        self._consumer_group = Conf.get(const.HA_GLOBAL_INDEX, f'CLUSTER_STOP_MON{_DELIM}consumer_group')
+        self._message_type = Conf.get(const.HA_GLOBAL_INDEX, f'CLUSTER_STOP_MON{_DELIM}message_type')
+        return MessageBus.get_consumer(consumer_id=self._consumer_id, \
```

Comment:
If topic (message type) is not yet created, will it throw errors at the beginning and then eventually start consuming messages once CSM is now up and has created the cluster_stop topic?
Also should this be named cluster_operations to allow any more manage operations on this path?

Reply:
Tested the behavior on Archana's setup, there is no issue in not-yet-existing topic consumer instance creation.
Also the name can remain cluster_stop as no other cluster operations are planned to be communicated between CSM and HA as yet, given most of the management will be done via k8s.